File: add/codesnippet/CSharp/m-system.net.dns.gethost_3_1.cs
   public void DisplayLocalHostName()
   {
      try {
         // Get the local computer host name.
         String hostName = Dns.GetHostName();
         Console.WriteLine("Computer name :" + hostName);
      }
      catch(SocketException e) {
         Console.WriteLine("SocketException caught!!!");
         Console.WriteLine("Source : " + e.Source);
         Console.WriteLine("Message : " + e.Message);
      }
	  catch(Exception e)
	  {
		  Console.WriteLine("Exception caught!!!");
		  Console.WriteLine("Source : " + e.Source);
		  Console.WriteLine("Message : " + e.Message);
	  }
   }

File: add/codesnippet/CPP/m-system.net.dns.gethost_3_1.cpp
public:
   void DisplayLocalHostName()
   {
      try
      {
         // Get the local computer host name.
         String^ hostName = Dns::GetHostName();
         Console::WriteLine( "Computer name : {0}", hostName );
      }
      catch ( SocketException^ e ) 
      {
         Console::WriteLine( "SocketException caught!!!" );
         Console::WriteLine( "Source : {0}", e->Source );
         Console::WriteLine( "Message : {0}", e->Message );
      }
      catch ( Exception^ e ) 
      {
         Console::WriteLine( "Exception caught!!!" );
         Console::WriteLine( "Source : {0}", e->Source );
         Console::WriteLine( "Message : {0}", e->Message );
      }
   }